base_interface.go

```diff
@@ -49,12 +57,38 @@ func requestNew(options options) rest.Request {
 		requestHeaders["On-Behalf-Of"] = options.Subuser
 	}
 
+	host, err := setDataResidency(options)
+	if err == nil {
+		options.Host = host
+	} else {
+		fmt.Println(err)
+		log.Println(err)
+	}
+
 	return rest.Request{
 		BaseURL: options.baseURL(),
 		Headers: requestHeaders,
 	}
 }
 
+func setDataResidency(options options) (string, error) {
+	currentHost := options.Host
+	defaultHost := allowedRegionsHostMap["global"]
+	if currentHost != defaultHost { // for testing, the hostname can be different
```

Comment:
What do you mean hostname can be different? For example if its `api.sendgrid.com` by default and I am calling setDataResidency('eu'), shouldn't the hostname change ?

Reply:
In the test cases, we set the URL of the fakeServer as the hostname (something like 127.0.0.1:5000) In that case if we return the default host (api.sendgrid.com) the test would fail. That's why I added this check.